dot/state/block.go

```diff
@@ -250,6 +250,37 @@ func (bs *BlockState) GetHashByNumber(num uint) (common.Hash, error) {
 	return common.NewHash(bh), nil
 }
 
+// GetBlocksBySlot gets all blocks that were produced in given slot.
+func (bs *BlockState) GetBlocksBySlot(slotNum uint64) ([]common.Hash, error) {
+	highestFinalisedHash, err := bs.GetHighestFinalisedHash()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get highest finalised hash: %w", err)
+	}
+
+	descendants, err := bs.bt.GetAllDescendants(highestFinalisedHash)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get descendants of %s: %w", highestFinalisedHash, err)
+	}
+
+	blocksWithGivenSlot := []common.Hash{}
```

Comment:
```suggestion
	blocksWithGivenSlot := make([]common.Hash, 0, len(descendants))
```

Reply:
I don't know if this is that necessary, if we only set one element in the slice, iterating from 100s of elements, it might not be worth setting a 100-sized capacity in memory. Unless most of the time most elements are appended in the slice then it's worth it. Not a big deal performance either I'd guess.